src/Datasource/EntityTrait.php

```diff
@@ -375,6 +375,87 @@ public function has($property)
     }
 
     /**
+     * Returns whether this entity contains a property named $property
+     * that contains a non-null value.
+     *
+     * ### Example:
+     *
+     * ```
+     * $entity = new Entity(['id' => 1, 'name' => null]);
+     * $entity->has('id'); // true
+     * $entity->has('name'); // false
+     * $entity->has('last_name'); // false
+     * ```
+     *
+     * You can check multiple properties by passing an array:
+     *
+     * ```
+     * $entity->has(['name', 'last_name']);
+     * ```
+     *
+     * All properties must not be null to get a truthy result.
+     *
+     * When checking multiple properties. All properties must not be null
+     * in order for true to be returned.
+     *
+     * @param string|array $property The property or properties to check.
+     * @return bool
+     */
+    public function has($property)
```

Comment:
Properties can have entities as values which closely mimics associations: `hasEntity()` and `hasEntities()` then.

Basically there should be:
- a way to check if a key property exists
- a way to check if a property is set to a non-nullish value
- a way to check if a property `is_a()` given Entity X or is a list with m...n items (or a closure for the count check)

Example:
```php
// property must exist but can be nullish
$entity->hasProperty('locale_code_user_preference');

// property must exist and cannot be nullish
$entity->hasValue('title');

// property must exist and must be is_a/subclass of Profile
$entity->hasEntity('profile', \App\Model\Profile::classname); 

// property must exist and returns true if the list contains 1 to 20 Comment entities. This 3rd param could also be a closure.
$entity->hasEntities('comments', \App\Model\Comment::classname, ['min' => 1, 'max' => '20']);

// ... or more general purpose, if preferred
$entity->hasObject('created', \Datetime::classname);
$entity->hasValue('created', \Datetime::classname);
```

For this PR then IMHO `hasValue()` + `hasProperty()` + deprecation of `has()` is good